### PR TITLE
Get btc price with websocket

### LINKED ISSUE #19: Check if performance gain can be achieved with different price querying
**Actual Behavior:**
Currently the bitcoin price is fetched once the first coin has been inserted.

**Expected Behavior:**
Might gain some response time if the price is fetched in a regular interval rather than after the first coin.

### PR DESCRIPTION
**Pull request by lth-elm**
Instead of fetching the btc price only when calls are made we open at the launch of the application a websocket that will run in a thread and regularly update the btc and satoshis price per usd(t) in a global variable so that we gain some response time. Unfortunately coingecko and coinmarketcap doesn't offer websocket solution so I used the binance one but it only refer prices with stablecoins which are mostly usd stablecoins and no other currencies.

Fixes #19

### CODE REVIEW

**Comment by lightisfaster** [comment]
Thank you for this nice feature. What do you think to have the "old" solution switchable in the config? I would test this PR the next day's.

File: config.py [hunk 1]
```diff
@@ -179,18 +179,19 @@ def create_config(config_file=None):
     logger.warning("No display configuration match.")
     sys.exit("Exiting...")
 
-# API URL for coingecko
-COINGECKO_URL_BASE = "https://api.coingecko.com/api/v3/"
+# API URL for binance websocket
+BINANCE_SOCKET = "wss://stream.binance.com:9443/ws/btcusdt@kline_1m" # every 2 seconds
```
Comment (lightisfaster) [inline config.py:183]: The "old" solution should be switchable in the config.
Many users want to use their own local currency

File: utils.py [hunk 2]
```diff
@@ -44,14 +47,32 @@ def create_font(font, size):
         print("Font not available")
 
 
-def get_btc_price(fiat_code):
-    """Get BTC -> FIAT conversion
+def get_btc_price():
+    """Get BTC Price in usd(t)
     """
-    url = config.COINGECKO_URL_BASE + "simple/price"
```
Comment (lightisfaster) [inline utils.py:50]: The "old" solution should be switchable in the config.
Many users want to use their own local currency

File: utils.py [hunk 3]
```diff
@@ -44,14 +47,32 @@ def create_font(font, size):
...
+
+
+    config.ws = websocket.WebSocketApp(config.BINANCE_SOCKET,
+                                on_open=on_open,
+                                on_message=on_message,
+                                on_close=on_close)
+                                
+    wst = threading.Thread(target=config.ws.run_forever)
+    wst.start()
+    time.sleep(2)
```
Comment (lightisfaster) [inline utils.py:75]: The update interval should be adjustable in the config.ini

**Review by lightisfaster — CHANGES_REQUESTED** [review]
Thanks for your PR.
The "old" solution should be switchable in the config because many users want to use their own local currency. The update interval should be adjustable in the config.ini, too.

**Comment by AxelHamburch** [comment]
I think this is a really very interesting option!

I tested that. Unfortunately this didn't work with the old image "2019-04-08-raspbian-stretch-lightningatm.gz". Running the command `pip3 install -r requirements.txt` does not find the required "websocket-client==1.2.3". It was only when I used a recent image that he was able to install the actual websocket client.

I discovered an error in app.py line 231:

`config.BTCPRICE = config.btc_price`

But that leads to an error. It should be:

`config.BTCPRICE = utils.get_btc_price(config.conf["atm"]["cur"])`

Then it works!

**Comment by AxelHamburch** [comment]
I think the image from 2019 with the "strech" os (V.9) is quite old and we should make a cut for the ATM and go to the latest version "bullseye" (V.11). When that's done we can also adjust the `requirements.txt` for this PR.

**Review by AxelHamburch — APPROVED** [review]

**Comment by AxelHamburch** [comment]
The program is therefore executable and we only have the problem that it is not downward compatible due to the requirements of `websocket-client==1.2.3` in requirements.txt. The OS with Debian Version 9 (Stretch) only know the `websocket-client` up to version 0.59.0. Unfortunately, it then makes a module import error when it starts. I haven't found a way to change it later. Only with the new version 11 (Bullseye) does it run smoothly.
  
We should consider how we can get a clean cut where we can also add future new features

2022-03-17 09_53_15-pi@raspberrypi_ ~_LightningATM
.

**Pull request closed without merging**